### PR TITLE
Fixes for unreachable project issues when transfer repository from organization

### PR DESCRIPTION
**Pull request by emrebdr**
When transferring repositories that have issues linked to a project board to another organization, the issues remain associated with the original project board. This causes the columns in the project board to become bugged, making it difficult to move other issues in or out of the affected columns. As a solution, I removed the issue relations since the other organization does not have this project table.

Fix for #31538

### CODE REVIEW

**Comment by emrebdr** [comment]
ready for review

File: services/repository/transfer.go [hunk 1]
```diff
@@ -177,6 +178,22 @@ func transferOwnership(ctx context.Context, doer *user_model.User, newOwnerName
 		}
 	}
 
+	// Remove project's issues that belong to old organization's projects
+	if oldOwner.IsOrganization() {
+		projects, err := project_model.GetAllProjectsIDsByOwnerIDAndType(ctx, oldOwner.ID, project_model.TypeOrganization)
```
Comment (lunny) [inline services/repository/transfer.go:183]: So then we need a warning popup window to warn that some project information will be removed but not deleting them silently.

**Review by lunny — CHANGES_REQUESTED** [review]

File: services/repository/transfer.go (hunk 1 above)
Comment (emrebdr) [reply services/repository/transfer.go:183]: Screenshot 2024-08-06 at 21 02 50

@lunny maybe I can add one more warning to this already existing warning modal. what do you think?

**Review by lunny — APPROVED** [review]

**Comment by emrebdr** [comment]
@wolfogre Hey do you have time for review please? 🙏

File: models/project/issue.go [hunk 2]
```diff
@@ -141,3 +141,15 @@ func (c *Column) moveIssuesToAnotherColumn(ctx context.Context, newColumn *Colum
...
+// DeleteAllProjectIssueByIssueIDsAndProjectIDs delete all project's issues by issue's and project's ids
+func DeleteAllProjectIssueByIssueIDsAndProjectIDs(ctx context.Context, issueIDs, projectIDs []int64) error {
+	for _, id := range projectIDs {
+		_, err := db.GetEngine(ctx).Where("project_id=?", id).In("issue_id", issueIDs).Delete(&ProjectIssue{})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```
Comment (wolfogre) [inline models/project/issue.go]: Why not
```suggestion
	_, err := db.GetEngine(ctx).In("project_id", projectIDs).In("issue_id", issueIDs).Delete(&ProjectIssue{})
	return err
```

Comment (emrebdr) [reply models/project/issue.go]: done 👍

**Review by wolfogre — APPROVED** [review]

**Comment by wolfogre** [comment]
Hmm, @GiteaBot doesn't backport it.

**Pull request merged**